test/test_tensorclass.py

```diff
@@ -2937,6 +2943,65 @@ def __setattr__(self, key, value):
         assert (s.b == 2).all()
 
 
+class TestTensorOnly:
```

Comment:
Should this be valid?
```python
    class NestedTensorOnly(TensorClass["tensor_only"]):
        a: torch.Tensor
        b: torch.Tensor | None
        c: TensorOnly
```
Either way, supported or not, should a test be added for this scenario?

Reply:
good catch, i think we can allow tensorclasses